src/routing/tests/mod.rs

```diff
@@ -1437,6 +1443,119 @@ async fn message_to_self(dst: MessageDst) -> Result<()> {
     Ok(())
 }
 
+#[tokio::test]
+async fn handle_elders_update() -> Result<()> {
+    // Start with section that has `ELDER_SIZE` elders with age 6, 1 non-elder with age 5 and one
+    // to-be-elder with age 7:
+    let node = create_node().with_age(MIN_AGE + 2);
+    let mut other_elder_peers: Vec<_> = iter::repeat_with(|| create_peer().with_age(MIN_AGE + 2))
+        .take(ELDER_SIZE - 1)
+        .collect();
+    let adult_peer = create_peer().with_age(MIN_AGE + 1);
+    let promoted_peer = create_peer().with_age(MIN_AGE + 3);
+
+    let sk_set0 = SecretKeySet::random();
+    let pk0 = sk_set0.secret_key().public_key();
+
+    let elders_info0 = EldersInfo::new(
+        iter::once(node.peer()).chain(other_elder_peers.clone()),
+        Prefix::default(),
+    );
+
+    let (mut section0, section_key_share) = create_section(&sk_set0, &elders_info0)?;
+
+    for peer in &[adult_peer, promoted_peer] {
+        let member_info = MemberInfo::joined(*peer);
+        let member_info = proven(sk_set0.secret_key(), member_info)?;
+        assert!(section0.update_member(member_info));
+    }
+
+    let demoted_peer = other_elder_peers.remove(0);
+
+    // Create `HandleConsensus` command for an `OurElders` vote. This will demote one of the
+    // current elders and promote the oldest peer.
+    let elders_info1 = EldersInfo::new(
+        iter::once(node.peer())
+            .chain(other_elder_peers.clone())
+            .chain(iter::once(promoted_peer)),
+        Prefix::default(),
+    );
+    let elder_names1: BTreeSet<_> = elders_info1.elders.keys().copied().collect();
```

Comment:
that's public keys collected?

Reply:
No, `XorName`s. `EldersInfo::elders` is a `BTreeMap<XorName, Peer>` so `elders.keys()` is an iterator of `XorName`s.